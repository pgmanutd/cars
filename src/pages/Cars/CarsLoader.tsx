import React from 'react';
import Box from '@material-ui/core/Box';

const CarsLoader: React.FC = () => (
  <Box component="section" data-testid="CarsLoader">
    Loading...
  </Box>
);

export default CarsLoader;
