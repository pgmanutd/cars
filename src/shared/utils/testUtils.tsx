import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory, MemoryHistory } from 'history';
import { render } from '@testing-library/react';

export const renderWithRouter = (
  ui: React.ReactElement,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  }: { route?: string; history?: MemoryHistory } = {},
) => ({
  ...render(<Router history={history}>{ui}</Router>),
  history,
});
