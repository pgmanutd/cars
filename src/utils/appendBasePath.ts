const appendBasePath = (basePath: string, path: string) =>
  `${basePath}?${path}`;

export default appendBasePath;
