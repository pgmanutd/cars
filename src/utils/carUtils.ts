import { Car } from 'features/CarInfo/types';

export const getCarName = ({ manufacturerName, modelName }: Car) =>
  `${manufacturerName} ${modelName}`;

export const getCarFeatures = (
  { stockNumber, mileage, fuelType, color }: Car,
  { stockNumberPrefix = '' }: { stockNumberPrefix?: string } = {
    stockNumberPrefix: '',
  },
) =>
  `${stockNumberPrefix}${stockNumber} - ${mileage?.number} ${mileage?.unit} - ${fuelType} - ${color}`;
