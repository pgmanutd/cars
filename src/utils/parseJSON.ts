const parseJSON = <T>(stringifiedJSON: string | null, fallbackValue: T) => {
  try {
    if (!stringifiedJSON) {
      return fallbackValue;
    }

    return JSON.parse(stringifiedJSON);
  } catch (e) {
    return fallbackValue;
  }
};

export default parseJSON;
