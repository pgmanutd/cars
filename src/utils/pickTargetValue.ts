import React from 'react';

const pickTargetValue = (event: React.ChangeEvent<{ value: unknown }>) =>
  event.target.value;

export default pickTargetValue;
