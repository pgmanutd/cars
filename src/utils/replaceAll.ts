export type ReplaceAllMappedObject = { [key: string]: string };

const replaceAll = (
  value: string,
  mappedObject: ReplaceAllMappedObject = {},
) => {
  const keys = Object.keys(mappedObject);

  if (!keys.length) {
    return value;
  }

  const regex = new RegExp(keys.join('|'), 'gi');

  return value.replace(regex, (matched) => {
    return mappedObject[matched];
  });
};

export default replaceAll;
