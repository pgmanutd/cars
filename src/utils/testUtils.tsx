import React from 'react';
import { Router, Route } from 'react-router-dom';
import { createMemoryHistory, MemoryHistory } from 'history';
import { HelmetProvider } from 'react-helmet-async';
import { render } from '@testing-library/react';

export const renderWithProviders = (
  ui: React.ReactElement,
  {
    routerConfig: {
      path = '/',
      route = '/',
      history = createMemoryHistory({ initialEntries: [route] }),
    },
  }: {
    routerConfig: { path?: string; route?: string; history?: MemoryHistory };
  } = { routerConfig: {} },
) => ({
  ...render(
    <HelmetProvider>
      <Router history={history}>
        <Route
          path={path}
          render={routeProps => React.cloneElement(ui, { ...routeProps })}
        />
      </Router>
    </HelmetProvider>,
  ),
  history,
});
