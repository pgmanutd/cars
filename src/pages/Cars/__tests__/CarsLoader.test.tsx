import React from 'react';

import { renderWithTranslate } from 'utils/testUtils';

import CarsLoader from '../CarsLoader';

describe('<CarsLoader />', () => {
  const setup = () => {
    const { renderResult } = renderWithTranslate(<CarsLoader />);

    return {
      renderResult,
    };
  };

  it('should render the component and matches it against stored snapshot', () => {
    const { renderResult } = setup();

    expect(renderResult.asFragment()).toMatchSnapshot();
  });
});
