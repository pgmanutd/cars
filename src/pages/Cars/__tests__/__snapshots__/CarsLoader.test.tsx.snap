// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarsLoader /> should render the component and matches it against stored snapshot 1`] = `
<DocumentFragment>
  <section
    class="MuiContainer-root makeStyles-root-1 MuiContainer-maxWidthLg"
    data-testid="CarsLoader"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
          style="height: 200px;"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
          style="height: 400px;"
        />
      </div>
    </div>
  </section>
</DocumentFragment>
`;
